src/iris/role_lookup/oncall.py

```diff
@@ -3,15 +3,21 @@
 
 import requests
 import logging
-from iris.metrics import stats
+from iris import metrics
 import oncallclient
 from iris.role_lookup import IrisRoleLookupException
 
 logger = logging.getLogger(__name__)
 
+# metrics
+stats_reset = {
+    'oncall_error': 0
+}
+
 
 class oncall(object):
     def __init__(self, config):
+        metrics.init(config, 'oncall-client', stats_reset)
```

Comment:
Do we ever call metrics.emit? 

It looks like the other uses of the metrics functionality in iris are part of the daemons rather than within the webapp uwsgi path.

It may make sense to create a single long lived "metrics" instance and gevent spawn metrics.emit_forever within get_api_app() or get_api()  so only one of them exists per uwsgi worker.

In an ideal scenario though we'd have a metrics aggregator  worker within uwsgi itself that would receive metrics updates from all of the uwsgi workers.

Reply:
@jrgp that's probably a better long term solution, reverting the metrics changes for now, will implement something like you suggested separately